precert/corr_test.go

```diff
@@ -0,0 +1,299 @@
+package precert
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestCorrespondIncorrectArgumentOrder(t *testing.T) {
+	pre, final, err := readPair("testdata/good/precert.pem", "testdata/good/final.pem")
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
We can use test.AssertError and test.AssertNotError for these

Reply:
I'd like to use the more traditional Go style `t.Fatal` / `t.Error` for these because (a) the distinction between fatal and error is actually expressive and can, for instance, allow gathering info on multiple failures at once, and (b) I'd like to contribute this code to certificate-transparency-go at some point.